fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/feature/PlanFeatures.java

```diff
@@ -0,0 +1,211 @@
+            if (vector != null) {
+                operatorVector.addAll(vector.finish());
+            } else {
+                operatorVector.addAll(AggregatedFeature.empty(opType));
+            }
+        }
+        this.operatorFeatureVectors = operatorVector;
+    }
+
+    public boolean skipOperator(OperatorType operatorType) {
+        if (EXCLUDE_OPERATORS.contains(operatorType)) {
+            return true;
+        }
+        /*
+          {@link OperatorFeatures.ScanOperatorFeatures}
+         */
+        if (operatorType.isPhysicalScan() && (operatorType != OperatorType.PHYSICAL_OLAP_SCAN)) {
+            return true;
+        }
+        return false;
+    }
+
+    public static Map<OperatorType, AggregatedFeature> aggregate(OperatorFeatures tree) {
+        Map<OperatorType, AggregatedFeature> sum = Maps.newHashMap();
+        aggregate(tree, sum);
+        return sum;
+    }
+
+    private static void aggregate(OperatorFeatures tree, Map<OperatorType, AggregatedFeature> sum) {
+        OperatorType opType = tree.opType;
+        AggregatedFeature exist = sum.computeIfAbsent(opType, (x) -> new AggregatedFeature(opType));
+        exist.merge(tree);
+
+        for (var child : tree.getChildren()) {
+            aggregate(child, sum);
+        }
+    }
+
+
+    private List<Long> extractTopTables() {
+        final int TOP_N_TABLES = 3;
+
+        List<Long> result = Lists.newArrayList();
+        for (int i = 0; i < TOP_N_TABLES; i++) {
+            result.add(0L);
+        }
+        if (CollectionUtils.isNotEmpty(tables)) {
+            result.addAll(tables.stream()
+                    .sorted(Comparator.comparing(x -> ((OlapTable) x).getRowCount()).reversed())
+                    .limit(TOP_N_TABLES)
+                    .map(Table::getId)
+                    .toList());
+            result = result.subList(result.size() - TOP_N_TABLES, result.size());
+        }
+        return result;
+    }
+
+    /**
+     * Aggregate vectors of same operators
+     */
+    public static class AggregatedFeature {
+        private final OperatorType opType;
+        private final List<Long> vector;
+        private int count = 0;
+
+        public AggregatedFeature(OperatorType type) {
+            this.opType = type;
+            int len = OperatorFeatures.vectorLength(type);
+            this.vector = Lists.newArrayListWithCapacity(len);
+            for (int i = 0; i < len; i++) {
+                this.vector.add(0L);
+            }
+        }
+
+        public void merge(OperatorFeatures node) {
+            this.count++;
+            // vector add
+            List<Long> vector1 = node.toVector();
+            for (int i = 0; i < vector.size(); i++) {
+                this.vector.set(i, this.vector.get(i) + vector1.get(i));
+            }
+        }
+
+        /**
+         * Build a real vector, with OPERATOR_TYPE and COUNT prefix
+         */
+        public List<Long> finish() {
+            List<Long> result = Lists.newArrayList();
```

Comment:
so agg operator contains features
- agg type
- how many agg operators
- sum of input rows?